### PR TITLE
fix #43 mac libs should be linked against @rpath

### LINKED ISSUE #43: Getting error while using Node v4.2.4
Hello~

Could you help me diagnose my error? Everything was working fine when I was using Node v 0.12.7.

```
return process.dlopen(module, path._makeLong(filename));
                 ^

Error: dlopen(/Users/angie/projects/node-projects/search-api/node_modules/node-libcurl/lib/binding/node_libcurl.node, 1): Library not loaded: /Users/travis/lib/libcurl.4.dylib
  Referenced from: /Users/angie/projects/node-projects/search-api/node_modules/node-libcurl/lib/binding/node_libcurl.node
  Reason: Incompatible library version: node_libcurl.node requires version 9.0.0 or later, but libcurl.4.dylib provides version 7.0.0
```

OS: OSX 10.10.5
xcode: 7.2
libcurl: 7.43.0

Thanks!

### PR DESCRIPTION
**Pull request by koskokos2**
This is fix for #43 mac. The best way how to deal with a problem of absolute paths embedded into binary is to replace the with @rpath, it requires list of paths where to look for libraries. Also I stopped linking against all libs libcurl depends on and link only against libcurl as this binary only depends on libcurl and libcurl itself handling its own dependencies just fine.

### CODE REVIEW

**Comment by simiansim** [comment]
@koskokos2 or @JCMais I think this line might need a semi colon:
CURL_PREFIX=/usr/local;
https://github.com/koskokos2/node-libcurl/blob/linking-improvement-mac/.travis.yml#L88

When run by travis...
```if [[ $TRAVIS_OS_NAME == "osx" ]]; then
   CURL_PREFIX=/usr/local
fi```
it look like this gets rewritten into
```if [[ $TRAVIS_OS_NAME == "osx" ]]; then CURL_PREFIX=/usr/local fi``
And so the bash "fi" is lost within the "then"

**Comment by koskokos2** [comment]
yeah right thanks, used to regular bash scripting and here that is just single line, so needs that semicolon

**Comment by JCMais** [comment]
I will merge this on this weekend and create a pre-release for it so it can be tested.

File: binding.gyp [hunk 1]
```diff
@@ -85,13 +85,19 @@
...
+                          '/usr/local/lib',
+                          '/usr/lib'
                         ]
                     },
                     'include_dirs' : [
                         '<!@(node "<(module_root_dir)/tools/curl-config.js" --cflags | sed s/-I//g)'
                     ],
                     'libraries': [
-                        '<!@(node "<(module_root_dir)/tools/curl-config.js" --libs)'
+                        '-L <!@(node "<(module_root_dir)/tools/curl-config.js" --prefix)/lib -lcurl'
```
Comment (JCMais) [inline binding.gyp]: @koskokos2 hi! I'm looking at some changes again, can you explain why that was changed? Was it causing issues with the linking on macOS?

Comment (koskokos2) [reply binding.gyp]: As far as I remember previous way caused prebuilt libs on Travis to have irrelevant dependencies when only -lcurl is needed.

Comment (JCMais) [reply binding.gyp]: makes sense, I'm adding support for static linking on linux and will need to put that back in, but will do so only on linux, will keep like the above for mac.

Thanks!

Comment (koskokos2) [reply binding.gyp]: Oh, that's amazing feature I was planning to start working on it myself in close future. Thanks for doing it!

**Pull request merged**